R/contingencytables.R

```diff
@@ -1057,67 +1061,79 @@ ContingencyTables <- function(jaspResults, dataset, options, ...) {
           CI          <- stats::confint(chi.result, level = level)
-          LogOR       <- unname(chi.result$coefficients)
-          log.CI.low  <- CI[1]
-          log.CI.high <- CI[2]
+          if (!options$LogOdds){
+            OR          <- exp(unname(chi.result$coefficients))
+            CI.low      <- exp(CI[1])
+            CI.high     <- exp(CI[2])
+          }else{
+            OR          <- unname(chi.result$coefficients)
+            CI.low      <- CI[1]
+            CI.high     <- CI[2]
+          }
         })
-
+        
         if (isTryError(chi.result))
           row[["value[oddsRatio]"]] <- NaN
         else if (is.na(chi.result))
           row[["value[oddsRatio]"]] <- NaN
         else {
-          row[["value[oddsRatio]"]] <- LogOR
-          row[["low[oddsRatio]"]]   <- log.CI.low
-          row[["up[oddsRatio]"]]    <- log.CI.high
+          row[["value[oddsRatio]"]] <- OR
+          row[["low[oddsRatio]"]]   <- CI.low
+          row[["up[oddsRatio]"]]    <- CI.high
           row[["p[oddsRatio]"]]     <- ""
         }
-        row[["value[oddsRatio]"]] <- LogOR
-        row[["low[oddsRatio]"]]   <- log.CI.low
-        row[["up[oddsRatio]"]]    <- log.CI.high
+        row[["value[oddsRatio]"]] <- OR
+        row[["low[oddsRatio]"]]   <- CI.low
+        row[["up[oddsRatio]"]]    <- CI.high
         row[["p[oddsRatio]"]]     <- ""
       }
     }
-
+    
     row[["type[FisherTest]"]] <- gettext("Fisher's exact test ")
     if (ready) {
-
+      
       if ( ! identical(dim(counts.matrix),as.integer(c(2,2)))) {
         row[["value[FisherTest]"]] <- NaN
         row[["low[FisherTest]"]]   <- ""
         row[["up[FisherTest]"]]    <- ""
         row[["p[FisherTest]"]]     <- ""
       } else {
         chi.result <- try({
-          conf.level  <- options$oddsRatioConfidenceIntervalInterval
+          conf.level  <- options$oddsRatioConfidenceIntervalIntervalNormal
           chi.result  <- stats::fisher.test(counts.matrix, conf.level = conf.level,
-                                            alternative = options$oddsRatioHypothesis)
-          OR          <- unname(chi.result$estimate)
-          logOR       <- log(OR)
-          log.CI.low  <- log(chi.result$conf.int[1])
-          log.CI.high <- log(chi.result$conf.int[2])
-          p           <- chi.result$p.value
+                                            alternative = options$oddsRatioHypothesisNormal)
+          if (!options$LogOdds){
+            OR          <- unname(chi.result$estimate)
+            CI.low      <- chi.result$conf.int[1]
+            CI.high     <- chi.result$conf.int[2]
+            p           <- chi.result$p.value
+          }else{
+            OR          <- log(unname(chi.result$estimate))
+            CI.low      <- log(chi.result$conf.int[1])
+            CI.high     <- log(chi.result$conf.int[2])
+            p           <- chi.result$p.value
+          }
```

Comment:
```suggestion
          if (options[["LogOdds"]]) {
            OR          <- log(unname(chi.result$estimate))
            CI.low      <- log(chi.result$conf.int[1])
            CI.high     <- log(chi.result$conf.int[2])
            p           <- chi.result$p.value
          } else {
            OR          <- unname(chi.result$estimate)
            CI.low      <- chi.result$conf.int[1]
            CI.high     <- chi.result$conf.int[2]
            p           <- chi.result$p.value
          }
```
always `[[` over `$` and watch the spacing around `else`. Also, `if (options[["LogOdds"]])` is preferred over `if (!options[["LogOdds"]])` if the only difference is flipping the two blocks.